### ISSUE TITLE
AdoptOpenJDK 14 OpenJ9 JRE archives contain `bin/javac`

### ISSUE THREAD

**Issue by breun** [AdoptOpenJDK/openjdk-support#81]
* **Platform and architecture (e.g. Linux/x64**: Linux/x64

* **Java version (from "java --version") e.g. 1.8.0_181, 11.0.4+11 etc.**: 14+36.1

* **Variant e.g. HotSpot or OpenJ9**: OpenJ9

* **Have you tested it with other vendors' java**: No

* **Steps to reproduce**: Download [AdoptOpenJDK 14+36.1 OpenJ9 JRE for Linux/x64](https://github.com/AdoptOpenJDK/openjdk14-binaries/releases/download/jdk-14%2B36.1_openj9-0.19.0/OpenJDK14U-jre_x64_linux_openj9_14_36_openj9-0.19.0.tar.gz) and [AdoptOpenJDK 14+36.1 OpenJ9XL JRE for Linux/x64](https://github.com/AdoptOpenJDK/openjdk14-binaries/releases/download/jdk-14%2B36.1_openj9-0.19.0/OpenJDK14U-jre_x64_linux_openj9_linuxXL_14_36_openj9-0.19.0.tar.gz) and extract the archives.

* **Error messsage and/or description of what is not working as expected**: The archives contain `bin/javac`, which I don't expect to find in a JRE archive. `bin/javac` also doesn't seem to present in the equivalent archives for Java 13.

**Comment by sxa555** [comment]
Appears to be the case for the Hotspot release too, and it wasn't the case for JDK13, so I'm going to adjust the title
```
[sxa@localhost shm]$ tar tvfz OpenJDK14U-jre_x64_linux_openj9_14_36_openj9-0.19.0.tar.gz  | grep javac-rwxr-xr-x root/root      8672 2020-03-18 19:31 jdk-14+36-jre/bin/javac
[sxa@localhost shm]$ tar tvfz OpenJDK14U-jre_x64_linux_hotspot_14_36.tar.gz | grep javac-rwxr-xr-x root/root      8672 2020-03-17 19:51 jdk-14+36-jre/bin/javac
[sxa@localhost shm]$ 
[sxa@localhost shm]$ tar tvfz OpenJDK13U-jre_x64_linux_hotspot_13.0.2_8.tar.gz | grep javac
[sxa@localhost shm]$ 
```

**Comment by keithc-ca** [comment]
I believe this chain leads to the inclusion of all of these (unexpected) launchers:
```
	javac
	javap
	jdeprscan
	jdeps
	jimage
	jlink
	jmod
	jpackage
	serialver
```

* `JRE_MODULES` includes `JRE_TOOL_MODULES` (see [1])
* `JRE_TOOL_MODULES` includes `jdk.incubator.jpackage` (see [2])
* `jdk.incubator.jpackage` requires `jdk.jlink` (see [3])
* `jdk.jlink` requires `jdk.jdeps` (see [4])
* `jdk.jdeps` requires `jdk.compiler` (see [5])
* `Launcher-jdk.compiler.gmk` defines the rules to build `javac` (see [6])

Perhaps this is in part why the make target `legacy-jre-image` includes the term 'legacy'.

[1] https://github.com/ibmruntimes/openj9-openjdk-jdk14/blob/openj9/make/Images.gmk#L46-L47
[2] https://github.com/ibmruntimes/openj9-openjdk-jdk14/blob/openj9/make/common/Modules.gmk#L135-L137
[3] https://github.com/ibmruntimes/openj9-openjdk-jdk14/blob/openj9/src/jdk.incubator.jpackage/share/classes/module-info.java#L49-L50
[4] https://github.com/ibmruntimes/openj9-openjdk-jdk14/blob/openj9/src/jdk.jlink/share/classes/module-info.java#L51-L53
[5] https://github.com/ibmruntimes/openj9-openjdk-jdk14/blob/openj9/src/jdk.jdeps/share/classes/module-info.java#L56-L58
[6] https://github.com/ibmruntimes/openj9-openjdk-jdk14/blob/openj9/make/launcher/Launcher-jdk.compiler.gmk#L28-L32

**Comment by karianna** [comment]
I'm not sure we'll want to change anything in that case?  The legacy target really is a best-effort target.

**Comment by keithc-ca** [comment]
I don't think we should expect much upstream help here. Instead of using the `legacy-jre-image` target, could use the following command to get something closer to what I think people will expect of a JRE:
```
images $ ./jdk/bin/jlink --add-modules java.se --no-header-files --no-man-pages --output jre
```
For OpenJ9, we will want to include the `openj9.jvm` and `openj9.sharedclasses` modules as well.

Thoughts, anyone?

**Comment by karianna** [comment]
It raises further questions about how we surface this to end-users.  It become 'our' definition off what an `AdoptOpenJDK JRE` is and isn't.  Flagging this for the TSC.
